### PR TITLE
refactor(api/nvim_cmd): use `kvec_t` for constructing cmdline string

### PR DESCRIPTION
**Pull request by famiu**

### CODE REVIEW

File: src/nvim/api/private/helpers.c [hunk 1]
```diff
@@ -1707,144 +1707,113 @@ bool string_iswhite(String str)
...
-    APPEND_MODIFIER("%dtab", cmdinfo->cmdmod.tab - 1);
+    kv_printf(cmdline, "%dtab ", cmdinfo->cmdmod.tab - 1);
   }
   if (cmdinfo->verbose != -1) {
-    APPEND_MODIFIER("%ldverbose", cmdinfo->verbose);
+    kv_printf(cmdline, "%ldverbose ", cmdinfo->verbose);
+  }
+
+  if (cmdinfo->emsg_silent) {
+    kv_printf(cmdline, "silent! ");
```
Comment (bfredl) [inline src/nvim/api/private/helpers.c:1727]: just use `kv_concat` when there's no formating

File: src/nvim/strings.c [hunk 2]
```diff
@@ -1480,3 +1480,32 @@ int vim_vsnprintf_typval(char *str, size_t str_m, const char *fmt, va_list ap, t
...
+
+  if (printed < 0) {
+    return -1;
+  }
+
+  // printed string didn't fit, resize and try again
+  if ((size_t)printed >= remaining) {
+    kv_ensure_space(*str, (size_t)printed + 1);  // include space for NUL terminator at the end
+    va_start(ap, fmt);
+    printed = vsnprintf(str->items + str->size, str->capacity - str->size, fmt, ap);
```
Comment (justinmk) [inline src/nvim/strings.c:1502]: coverity:

```
*** CID 352782:  Null pointer dereferences  (FORWARD_NULL)
/src/nvim/strings.c: 1502 in kv_do_printf()
1496       }
1497     
1498       // printed string didn't fit, resize and try again
1499       if ((size_t)printed >= remaining) {
1500         kv_ensure_space(*str, (size_t)printed + 1);  // include space for NUL terminator at the end
1501         va_start(ap, fmt);
>>>     CID 352782:  Null pointer dereferences  (FORWARD_NULL)
>>>     Passing null pointer "str->items + str->size" to "vsnprintf", which dereferences it.
1502         printed = vsnprintf(str->items + str->size, str->capacity - str->size, fmt, ap);
1503         va_end(ap);
1504         if (printed < 0) {
1505           return -1;
1506         }
1507       }
1508     
1509       str->size += (size_t)printed;
1510       return printed;

Comment (famiu) [reply src/nvim/strings.c:1502]: @bfredl This is strange, isn't `kv_ensure_space` supposed to ensure that `str->items` isn't a NULL pointer?

Comment (bfredl) [reply src/nvim/strings.c:1502]: yes. this expands to something with `xrealloc` which is `FUNC_ATTR_NONNULL_RET` . so a plain fuck-up by coverity.

Comment (dundargoc) [reply src/nvim/strings.c:1502]: @jamessan ignore this error too pls

Comment (jamessan) [reply src/nvim/strings.c:1502]: It has nothing to do with `xrealloc`.  The problem is that line 1491 checks if `str->items` is null, which therefore tells Coverity it _can_ be NULL.  Then line 1502 doesn't do that check, so Coverity says that's a problem.  Nothing in `kv_ensure_size()` is doing a NULL check, so Coverity has no reason to think it's changing `str->items`.

Comment (jamessan) [reply src/nvim/strings.c:1502]: Putting `assert(str->items);` before line 1502 would indicate we know it's non-null.

**Pull request merged**